lib/api/dsl/storage/storeOperands.js

```diff
@@ -17,344 +38,344 @@ const
+   * @param {object} subfilter
+   * @param {object} condition
    */
-  Object.keys(args).forEach(a => {
-    if (['gt', 'gte'].indexOf(a) !== -1) {
-      low = a === 'gte' ? args[a] - 1e-10 : args[a];
+  notrange(foPairs, subfilter, condition) {
+    const
+      field = Object.keys(condition.value)[0],
+      args = condition.value[field];
+    let
+      low = -Infinity,
+      high = Infinity;
+
+    /*
+     Initializes low and high values depending on condition arguments
+     As the interval tree library used only considers inclusive boundaries,
+     we need to add or substract an epsilon value to provided arguments
+     for lte and gte options
+     This is the reverse operation than the one done for the "range"
+     keyword, as we then invert the searched range.
+     */
+    Object.keys(args).forEach(a => {
+      if (['gt', 'gte'].indexOf(a) !== -1) {
+        low = a === 'gte' ? args[a] - 1e-10 : args[a];
+      }
+
+      if (['lt', 'lte'].indexOf(a) !== -1) {
+        high = a === 'lte' ? args[a] + 1e-10 : args[a];
+      }
+    });
+
+    if (!foPairs.fields[field]) {
+      foPairs.keys.insert(field);
+      foPairs.fields[field] = {
+        tree: new IntervalTree(),
+        count: 1,
+        subfilters: {
+          [subfilter.id]: {subfilter, low, high}
+        }
+      };
+    }
+    else {
+      foPairs.fields[field].subfilters[subfilter.id] = {subfilter, low, high};
+      foPairs.fields[field].count++;
     }
 
-    if (['lt', 'lte'].indexOf(a) !== -1) {
-      high = a === 'lte' ? args[a] + 1e-10 : args[a];
+    if (low !== -Infinity) {
+      foPairs.fields[field].tree.insert(-Infinity, low, subfilter);
     }
-  });
-
-  if (!foPairs.fields[field]) {
-    foPairs.keys.insert(field);
-    foPairs.fields[field] = {
-      tree: new IntervalTree(),
-      count: 1,
-      subfilters: {
-        [subfilter.id]: {subfilter, low, high}
-      }
-    };
-  }
-  else {
-    foPairs.fields[field].subfilters[subfilter.id] = {subfilter, low, high};
-    foPairs.fields[field].count++;
-  }
 
-  if (low !== -Infinity) {
-    foPairs.fields[field].tree.insert(-Infinity, low, subfilter);
+    if (high !== Infinity) {
+      foPairs.fields[field].tree.insert(high, Infinity, subfilter);
+    }
   }
 
-  if (high !== Infinity) {
-    foPairs.fields[field].tree.insert(high, Infinity, subfilter);
+  /**
+   * Stores a "regexp" condition into the field-operand structure
+   *
+   * @param {object} foPairs
+   * @param {object} subfilter
+   * @param {object} condition
+   */
+  regexp(foPairs, subfilter, condition) {
+    const
+      fieldName = Object.keys(condition.value)[0],
+      value = new RegexpCondition(condition.value[fieldName].value, subfilter, condition.value[fieldName].flags);
+    let idx;
+
+    if (!foPairs.fields[fieldName]) {
+      foPairs.keys.insert(fieldName);
+      foPairs.fields[fieldName] = {
+        expressions: new SortedArray([value], (a, b) => strcmp(a.stringValue, b.stringValue))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 93 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)